docs/cart-and-checkout-blocks/removing-checkout-fields.md

```diff
@@ -0,0 +1,197 @@
+---
+post_title: Cart and Checkout - Removing checkout fields
+menu_title: Extensibility
+tags: reference
+---
+<!-- markdownlint-disable MD041 -->
+
+If you're trying to extend the new Checkout block, you might notice the previous `checkout_fields` isn't working. This is intentional, we're taking back control of how to render and handle fields, yet, one of the most common requests we get is how to disable Checkout fields for certain order types. For a long time, our answer was "you can't". You actually can, but we don't encourage it.
+
+## Why we don't encourage removing fields
+
+The simple reason is that Checkout is complex, and many plugins need different data, payment methods will need your address info to run fraud detection, or validate your card, tax (including core) will need your full address to calculate taxes correctly, this is something we still think about in Checkout and are looking for a universal solution, until then, you can go with code, and here's how:
+
+## Disabling fields for a single country
+
+Checkout fields still country locale, which means you can modify fields for a country via PHP and this would work fine. The following billing form has 10 fields, 7 of them are required:
+
+![Image](https://github.com/user-attachments/assets/63d83769-c20c-4c85-aebf-da8510d1d9ae)
+
+Let's say that in Algeria, we know for sure that postal code and city are redundant, so we can remove them:
+
+```php
+add_filter('woocommerce_get_country_locale', function( $locale ) {
+
+	$locale['DZ']['postcode']['required'] = false;
+	$locale['DZ']['postcode']['hidden'] = true;
+
+	$locale['DZ']['city']['required'] = false;
+	$locale['DZ']['city']['hidden'] = true;
+
+		return $locale;
+});
+```
+
+With the code above, both of those fields are removed for Algeria, but not for other countries, which is a good and safe update.
+
+![Image](https://github.com/user-attachments/assets/96d45e1a-99f0-4b91-92d1-85ee742a9705)
+
+Keep in mind that this will remove fields for both shipping and billing. We think strongly that billing shape should match shipping shape.
+
+## Removing every field except Country and names
+
+We can follow up with removing all other fields except Country, removing the country would fail your order, as we need country to know which fields to remove:
+
+```php
+add_filter('woocommerce_get_country_locale', function( $locale ) {
+
+	$locale['DZ']['address_1'] = [
+		'required' => false,
+		'hidden'   => true,
+	];
+
+	$locale['DZ']['postcode'] = [
+		'required' => false,
+		'hidden'   => true,
+	];
+
+	$locale['DZ']['city'] = [
+		'required' => false,
+		'hidden'   => true,
+	];
+
+	$locale['DZ']['state'] = [
+		'required' => false,
+		'hidden'   => true,
+	];
```

Comment:
```suggestion
	$locale['DZ']['company'] = [
		'required' => false,
		'hidden'   => true,
	];
	
	$locale['DZ']['state'] = [
		'required' => false,
		'hidden'   => true,
	];
	
	$locale['DZ']['phone'] = [
		'required' => false,
		'hidden'   => true,
	];
```

Since published, we can now remove phone and company via PHP, I added them here.